### PR TITLE
Add <sys/timex.h> bindings for linux-musl

### PR DESCRIPTION
**Pull request by dfoxfranke**
Adds bindings for `adjtimex()` & co. for linux-musl. Also adds `clock_adjtime()` to linux-glibc.

### CODE REVIEW

**Comment by MarkAtwood** [comment]
AWS OSPO review, no OSPO concerns.

File: src/unix/linux_like/linux/musl/mod.rs [hunk 1]
```diff
@@ -249,6 +249,36 @@ s! {
...
+        pub ppsfreq: ::c_long,
+        pub jitter: ::c_long,
+        pub shift: ::c_int,
+        pub stabil: ::c_long,
+        pub jitcnt: ::c_long,
+        pub calcnt: ::c_long,
+        pub errcnt: ::c_long,
+        pub stbcnt: ::c_long,
+        pub tai: ::c_int,
+        pub __padding: [::c_int; 11],
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/musl/mod.rs:274]: A padding field doesn't need to be public generally, is there any reason?

Comment (dfoxfranke) [reply src/unix/linux_like/linux/musl/mod.rs:274]: I suppose it doesn't matter much, but if this one should be private then I should've made the [glibc](https://docs.rs/libc/0.2.99/libc/struct.timex.html) ones private as well when I contributed #1642. So I can either keep this one public, be inconsistent, or make a breaking change to the glibc version. Up to you, I have no strong feelings on the matter.

Comment (JohnTitor) [reply src/unix/linux_like/linux/musl/mod.rs:274]: Ah makes sense! Let's keep it as-is for now.

**Review by JohnTitor — APPROVED** [review]

**Comment by dfoxfranke** [comment]
Looks like a CI test is failing because <http://ftp.debian.org/debian/dists/testing/main/installer-s390x/20200314/images/generic/initrd.debian> has gone 404 (maybe releated to the release of Debian 11)? Obviously unrelated to this PR.

**Comment by JohnTitor** [comment]
Fixed on master, @bors retry

**Pull request merged**